WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/ProductReviewsViewModelTests.swift

```diff
@@ -0,0 +1,172 @@
+import XCTest
+@testable import WooCommerce
+@testable import Networking
+@testable import Yosemite
+
+final class ProductReviewsViewModelTests: XCTestCase {
+    private var mockDataSource: ReviewsDataSource!
+    private var viewModel: ProductReviewsViewModel!
+    private let productID: Int64 = 12345
+
+    override func setUp() {
+        super.setUp()
+        mockDataSource = MockProductReviewsDataSource()
+        viewModel = ProductReviewsViewModel(data: mockDataSource)
+    }
+
+    override func tearDown() {
+        viewModel = nil
+        mockDataSource = nil
+        super.tearDown()
+    }
+
+    func test_dataSource_returns_injected_ProductReviewsDataSource() {
+        let dataSource = viewModel.dataSource
+        XCTAssertNotNil(dataSource as? MockProductReviewsDataSource)
+    }
+
+    func test_delegate_returns_injected_ProductReviewsDelegate() {
+        let delegate = viewModel.delegate
+        XCTAssertNotNil(delegate as? MockProductReviewsDataSource)
+    }
+
+    func test_isEmpty_returns_the_same_as_the_dataSource() {
+        XCTAssertEqual(viewModel.isEmpty, mockDataSource.isEmpty)
+    }
+
+    func test_display_placeHolder_reviews_stops_forwarding_events_in_dataSource() {
+        let table = UITableView()
+        let ds = mockDataSource as! MockProductReviewsDataSource
+
+        viewModel.displayPlaceholderReviews(tableView: table)
+
+        XCTAssertTrue(ds.stopsForwardingEventsWasHit)
+    }
+
+    func test_remove_placeHolder_reviews_starts_forwarding_events_in_dataSource() {
+        let table = UITableView()
+        let ds = mockDataSource as! MockProductReviewsDataSource
+
+        viewModel.removePlaceholderReviews(tableView: table)
+
+        XCTAssertTrue(ds.startForwardingEventsWasHit)
+    }
+
+    func test_configure_resultsController_starts_forwarding_events_and_starts_observing_reviews_in_dataSource() {
+        let table = UITableView()
+        let ds = mockDataSource as! MockProductReviewsDataSource
+
+        viewModel.configureResultsController(tableView: table)
+
+        XCTAssertTrue(ds.startForwardingEventsWasHit && ds.startObservingWasHit)
+    }
+
+    func test_sync_data_hits_expected_reviews_and_products_actions() {
```

Comment:
nit: just a suggestion to structure the test more like how we usually do Given-When-Then :)